### ISSUE TITLE
Crashes if too much memory is allocated

### ISSUE THREAD

**Issue by hexahigh** [nepx/halfix-demo#2]
I tried allocating 500mb of memory and 52mb of video ram.
I know that the issue is not because of the Disk image as it boots correctly if i give it less ram.
I tested this on a chromebook and on a computer with a intel i5 and a rtx 3060.
I tested it on chrome and firefox, both giving the same result.
The disk image was a Android-x86 32 bit disk image

Below you can find the disk image and the log file.

Log file:
[nepx.github.io-1673607609684.log](https://github.com/nepx/halfix-demo/files/10410868/nepx.github.io-1673607609684.log)

Disk image:
Link 1. https://pomf2.lain.la/f/1p85z7qw.iso
Link 2. https://mirrorace.org/m/3a0OR

**Comment by nepx** [comment]
It looks like it's running into an out of memory exception. I've been playing around with Emscripten, but it's been frustratingly difficult to force the runtime to allocate more memory. I think that compiling directly using Clang would fix this issue since we can specify memory sizes directly, but that would require us to move away from SDL. I would have to do this anyway if it's going to run in a Web Worker, but it would take some work. 

I tested your disk image. 128 MB of RAM on the native version but Android will crash during boot due to an unimplemented SSSE3 opcode (it's `66 0F 38 02`, `PHADDD`). I'm working on a rewrite of the emulator right now to support 64-bit and add a dynamic recompiler, and I will test your disk image at some point. 

TL;DR: out of memory exception. I couldn't find any good way to allow larger RAM sizes without allocating more memory up front, which would crash lower-end phone browsers. I will have to look at this in greater detail when I port the rewrite to JavaScript.

**Comment by hexahigh** [comment]
Alright, thanks for your reply.
